src/main/java/Duke.java

```diff
@@ -1,10 +1,152 @@
+import util.Parser;
+import util.Storage;
+import util.Ui;
+
+import task.TaskList;
+import task.Event;
+import task.Deadline;
+import task.Todo;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.Scanner;
+
 public class Duke {
-    public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+    private File newFile;
+    private Parser parser;
+
+
+    public Duke(String filePath) {
+        ui = new Ui();
+        storage = new Storage(filePath);
+        newFile = new File(filePath);
+        tasks = new TaskList();
+        parser = new Parser();
+
+        try {
+            if (newFile.exists()) {
+                storage.loadFile(tasks.list);
+            }
+        } catch (FileNotFoundException e) {
+            ui.reply("Please create the text file data/duke.txt");
+        }
+
+
+
+    }
+
+    public void run() {
+        Scanner sc = new Scanner(System.in);
+
+        ui.greet();
+
+        String tab = "    ";
+
+        parser.parse(sc.nextLine());
+        String task = parser.getTask();
+        String item = parser.getItem();
+
+
+        while (!task.equals("bye")) {
```

Comment:
Each string here can be split into separate methods to improve readability of the code. 

Example:
public void deadlineTask() {
         if (!item.equals("")) {
                    String thing = item.split(" /by ")[0];
                    String time = item.split(" /by ")[1];
                    tasks.add(new Deadline(thing, time));
         } else {
                    ui.reply("Can read instructions or not? Deadline cannot be empty :/");
         }
}

Thus inside this while loop it is just:
while ...
            } else if (task.equals("deadline")) {
                          deadlineTask();
           }
...